src/Schema.php

```diff
@@ -121,7 +121,18 @@ private static function createOrPrependFile($pathname, $template, $separator = '
     else {
       $content = file_get_contents($pathname);
       if (FALSE === strpos($content, $template)) {
-        $content = $template . $separator . $content;
+        // Remove old content if present, looking for the template block's
+        // BEGIN and END lines.
+        $template_lines = explode("\n", rtrim($template, "\n"));
+        $begin = strpos($content, $template_lines[0]);
+        $lastLine = end($template_lines);
+        $end = strpos($content, $lastLine) + strlen($lastLine);
+        if ($begin !== FALSE && $end !== FALSE) {
+          $content = substr_replace($content, $template, $begin, $end - $begin + 1);
```

Comment:
I still don't really understand how this works; I would simply do it with a preg_replace() on the $content instead… but I trust that you tested it extensively (and because you understand it, you can maintain it in the future 😀)